packages/sit-onyx/src/composables/useVModel.ts

```diff
@@ -0,0 +1,34 @@
+import { computed, ref } from "vue";
+export type Nullable<T> = T | undefined | null;
+export type UseVModelOptions<
+  TKey extends keyof TProps & string,
+  TProps extends object,
+  TValue extends TProps[TKey],
+  TDefaultValue = Nullable<TValue>,
+> = {
+  key: TKey;
+  props: TProps;
+  emit: TValue extends undefined | null
+    ? (evt: `update:${TKey}`, value?: Nullable<TValue>) => void
+    : (evt: `update:${TKey}`, value: TValue) => void;
+  defaultValue?: TDefaultValue;
```

Comment:
We need to use a factory function, so that non-primitive values can be mutated without issues.
```suggestion
  defaultValue?: () => TDefaultValue;
```